src/stores/geometryEventStore.ts

```diff
@@ -0,0 +1,103 @@
+import { action, computed, observable, observe, IObjectDidChange } from 'mobx';
+import IEvent from '~/models/IEvent';
+import eventTypes from '~/enums/eventTypes';
+import entityNames from '~/enums/entityNames';
+import { IRoutePathLink } from '~/models';
+import RoutePathStore from './routePathStore';
+import Navigator from '../routing/navigator';
+
+const enum IObjectDidChangeUpdateTypes {
+    ADDED = 'added',
+}
+
+export class GeometryEventStore {
+    @observable private _events: IEvent[];
+
+    constructor() {
+        this._events = [];
+
+        this.initRoutePathLinkObservable();
+        this.initClearEventListOnPageChange();
+    }
+
+    @computed get events(): IEvent[] {
+        return this._events;
+    }
+
+    @action
+    private clearEvents() {
+        this._events = [];
+    }
+
+    @action
+    private pushToEvents(
+        { action, entityName, objectId, oldObject, newObject }
+      : {
+          action: eventTypes,
+          entityName: entityNames,
+          objectId: string,
+          oldObject?: object,
+          newObject?: object,
+      },
+      ) {
+        this._events.push({
+            action,
+            objectId,
+            postObject: newObject,
+            preObject: oldObject,
+            timestamp: new Date(),
+            entity: entityName,
+        });
+    }
+
+    private logRoutePathLinkChanges(change: IObjectDidChange) {
+        if (change.hasOwnProperty(IObjectDidChangeUpdateTypes.ADDED)) {
+            change[IObjectDidChangeUpdateTypes.ADDED].slice()
+                .forEach((routePathLink: IRoutePathLink) => {
+                    this.pushToEvents({
+                        action: eventTypes.ADD,
+                        entityName: entityNames.ROUTELINK,
+                        newObject: routePathLink,
+                        objectId: routePathLink.id,
+                    });
+                });
+        }
+    }
+
+    private initClearEventListOnPageChange() {
```

Comment:
Hmm, how about having to call this manually in NewRoutePath's componentWillUnmount and rename it as removeEvents()? I wouldn't make an asumption that we would always want to reset eventStore when sidebar page changes.

Reply:
We have the possibility to modify the clear calling later, if we find the need for this. But I would suggest keeping it isolated as long as we dont see the need for extracting it.